### PR TITLE
[g2o] Update to 20230806_git

### LINKED ISSUE #42507: [g2o] update to 20230223_git
### Library name

g2o

### New version number

20230223_git

### Other information that may be useful (release notes, etc...)

3 years of updates

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/42507
Fixed the problem of including absolute paths in config.h file.
Remove patch csparse.diff.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Usage test passed with x64-windows triplet.

### CODE REVIEW

File: ports/g2o/fix-absolute.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c3b6ab5..2bad328 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -487,7 +487,6 @@ set(G2O_HAVE_CHOLMOD ${CHOLMOD_FOUND})
+ set(G2O_HAVE_CSPARSE ${G2O_USE_CSPARSE})
+ set(G2O_SHARED_LIBS ${BUILD_SHARED_LIBS})
+ set(G2O_LGPL_SHARED_LIBS ${BUILD_LGPL_SHARED_LIBS})
+-set(G2O_CXX_COMPILER "${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER}")
```
Comment (dg0yt) [inline ports/g2o/fix-absolute.patch:9]: Is it necessary to remove`${CMAKE_CXX_COMPILER_ID}`?

File: ports/g2o/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "g2o",
-  "version-date": "2020-02-07",
-  "port-version": 6,
+  "version-string": "20230806_git",
```
Comment (dg0yt) [inline ports/g2o/vcpkg.json:3]: Do not use `version-string`. (Mind `version>=` in manifests.)
~~~suggestion
  "version-date": "2023-08-06",
~~~

File: ports/g2o/portfile.cmake [hunk 3]
```diff
@@ -1,11 +1,11 @@
 vcpkg_from_github(
```
Comment (dg0yt) [inline ports/g2o/portfile.cmake:1]:
```suggestion
string(REPLACE "-" "" GIT_TAG "${VERSION}_git")
vcpkg_from_github(
```

File: ports/g2o/portfile.cmake [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
     OUT_SOURCE_PATH SOURCE_PATH
     REPO RainerKuemmerle/g2o
-    REF fcba4eaca6f20d9a5792404cc8ef303aeb8ba5d2
-    SHA512 41e5a6d40db10d66182653d56f937f29264bf4f9412dfa651be949caeb055741c9d9ba75a122180892aafe7d45b334d50470284121148c0561e1d49f6ba5e20a
+    REF "${VERSION}"
```
Comment (dg0yt) [inline ports/g2o/portfile.cmake:4]:
```suggestion
    REF "${GIT_TAG}"
```

File: ports/g2o/fix-absolute.patch (hunk 1 above)
Comment (jimwang118) [reply ports/g2o/fix-absolute.patch]: Removed from patch.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**